### ISSUE TITLE
x-tagGroups Only keeps first file's tags

### ISSUE THREAD

**Issue by pedrodagraca** [robertmassaioli/openapi-merge#60]
Hi,

I'm using the cli to merge 2 openapi json files. The x-tagGroups in the result json only has the first file's x-tagGroups array.

Any ideas?

**Comment by robertmassaioli** [comment]
This is expected behaviour according to the test cases: https://github.com/robertmassaioli/openapi-merge/blob/main/packages/openapi-merge/src/__tests__/x-tensions.test.ts

The `x-*` fields are extensions of OpenAPI and could be in any format at all. Is this a feature request for merging arbitrary JSON blobs in `x-*` fields? Maybe for being able to provide you own custom merge logic just for that field?

**Comment by pedrodagraca** [comment]
I see. Yes, in that case this would be a feature request for merging those x-* blobs. I currently use Redoc, which uses the x-tagGroups field in order to create a higher level grouping of the endpoints.

**Comment by WenceslaoMateos** [comment]
I think that the program should treat the x-* tags the same as the others, because when you use some implementation, (e.g: redocly with the x-tagGroups) the implmentation itself has consistency. It could be a really useful feature for the future. (I'm currently having the same problem as @pedrodagraca)

**Comment by robertmassaioli** [comment]
I'm thinking that the best way to handle this would be the ability to provide custom merging functions that only run on `x-` tags. As a result, i'm not sure if it would work with the CLI but I'll see if I can think of something. The initial target should probably be at the library first.

**Comment by robertmassaioli** [comment]
Superseded by #153, which generalizes this to a configurable `extensionMergeStrategies` option covering **any** `x-*` extension via a recursive merge-strategy tree, not just `x-tagGroups`. Design rationale in [proposal 48](https://github.com/robertmassaioli/openapi-merge/blob/main/ai-planning/48-proposal-configurable-extension-merge-strategies.md).

The behaviour this issue asked for — combining `x-tagGroups` across inputs instead of first-wins — is fully reproducible as configuration:

```json
"extensionMergeStrategies": {
  "x-tagGroups": {
    "kind": "array",
    "strategy": "union-by-key",
    "key": "name",
    "item": {
      "kind": "object",
      "strategy": "merge",
      "fields": {
        "tags": { "kind": "array", "strategy": "concat-unique" }
      }
    }
  }
}
```

Groups sharing a `name` combine their tags (deduplicated); a group seen only once passes through unchanged; both appear in first-seen order. One thing this does *not* reproduce from #127's approach: pruning a group left with zero tags after merging — a known, documented gap (see proposal 48 §4/§7), since that's a predicate on the merged result rather than a policy for combining two inputs' values.

Docs: [Library reference → `extensionMergeStrategies`](https://robertmassaioli.github.io/openapi-merge/library/merge-options#extensionmergestrategies) · [CLI examples → Recreating `x-tagGroups`'s merge as configuration](https://robertmassaioli.github.io/openapi-merge/cli/examples#recreating-x-taggroups-s-merge-as-configuration)

Closing as resolved by the general mechanism rather than a `x-tagGroups`-specific fix.
